.github/workflows/generate-bump-pages.yml

```
@@ -0,0 +1,73 @@
+name: Check & deploy API documentation
+
+on:
+  workflow_dispatch: # Allow manual trigger in case of quick fix
+  push:
+    branches:
+      - main
+    paths:
+      - 'openapi/**.json'
+
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'openapi/**.json'
+
+permissions:
+  contents: read
+
+jobs:
+  create-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+          spec_mapping=$(node .github/scripts/generateSpecMapping.js)
+          echo "matrix=$spec_mapping" >> "$GITHUB_OUTPUT"
+
+  deploy-doc:
+    needs: create-matrix
+    if: ${{ github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
+    name: Deploy API documentation on Bump.sh
+    strategy:
+      matrix: 
+        spec-mapping: ${{ fromJson(needs.create-matrix.outputs.matrix) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Deploy API documentation
+        uses: bump-sh/github-action@v1
```

FYI: We ping to a gitsha when the third party gh action is not from github
```suggestion
        uses: bump-sh/github-action@690c81156715f37cb72d006e5cbb81fbd9b45365
```